### PR TITLE
Feature/taipy#126 group taipy cli args to subcommands

### PR DESCRIPTION
**Pull request by trgiangdo**
In this PR:
- Change the behavior (or a bug) where the arguments loaded from the toml file will override CLI arguments. It should be the other way around.
- Add loading args from TOML and CLI test case.

### CODE REVIEW

File: tests/taipy/gui/cli/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+    # Override Config.configure_gui() arguments by loading a TOML file with [gui] section
+    toml_config = NamedTemporaryFile(
+        content="""
+[TAIPY]
+
+[gui]
+host = "my_3rd_host"
+port = 5003
+use_reloader = "true:bool"
```
Comment (FabienLelaquais) [inline tests/taipy/gui/cli/test_cli.py:140]: I don't understand what that is not simply
`use_reloader = true`

That would be very valid TOML...

File: src/taipy/gui/_gui_cli.py [hunk 2]
```diff
@@ -9,15 +9,26 @@
...
+        "no_debug": False,
+        "use_reloader": False,
+        "no_reloader": False,
+    }
 
     @classmethod
-    def _create_parser(cls):
-        gui_parser = _Argparser._add_groupparser("GUI", "Optional arguments for GUI service")
+    def create_parser(cls):
+        gui_parser = _CLI._add_subparser("taipy", help="Run application with Taipy arguments.")
```
Comment (FabienLelaquais) [inline src/taipy/gui/_gui_cli.py]: This I think breaks the legacy parsing mechanism.
Users would for example use `python script.py -debug` and get what they expect.
Now it seems that forces them to use `python script.py taipy -debug`.

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Please make sure my worry is addressed.

File: src/taipy/gui/_gui_cli.py (hunk 2 above)
Comment (FabienLelaquais) [reply src/taipy/gui/_gui_cli.py]: What I'm saying is that we need to rethink this thing and try to allow not to use the `taipy` option

File: tests/taipy/gui/cli/test_cli.py (hunk 1 above)
Comment (trgiangdo) [reply tests/taipy/gui/cli/test_cli.py:140]: In taipy-config, we are using the `:bool` to separate a boolean from a string.

This is just how it was implemented.

**Review by FredLL-Avaiga — APPROVED** [review]

File: src/taipy/gui/_gui_cli.py [hunk 3]
```diff
@@ -9,15 +9,15 @@
...
 
 
 class _GuiCLI:
     """Command-line interface of GUI."""
 
     @classmethod
-    def _create_parser(cls):
-        gui_parser = _Argparser._add_groupparser("GUI", "Optional arguments for GUI service")
+    def create_parser(cls):
+        gui_parser = _CLI._add_groupparser("Taipy GUI", "Optional arguments for Taipy GUI service")
```
Comment (FabienLelaquais) [inline src/taipy/gui/_gui_cli.py:20]: What's the point of the 'Taipy ' prefix?

**Review by FabienLelaquais — APPROVED** [review]
Looks good to me!

File: src/taipy/gui/_gui_cli.py (hunk 3 above)
Comment (trgiangdo) [reply src/taipy/gui/_gui_cli.py:20]: Just that when we show the help message, it is clear that these options are for Taipy GUI

**Pull request merged**